mrblib/string.rb

```diff
@@ -154,13 +154,48 @@ def each_byte(&block)
   end
 
   ##
-  # Modify +self+ by replacing the content of +self+
-  # at the position +pos+ with +value+.
-  def []=(pos, value)
-    pos += self.length if pos < 0
-    b = self[0, pos]
-    a = self[pos + 1..-1]
-    self.replace([b, value, a].join(''))
+  # Modify +self+ by replacing the content of +self+.
+  # The portion of the string affected is determined using the same criteria as +String#[]+.
+  def []=(*args)
+    anum = args.size
+    if anum == 2
+      pos, value = args
+      if pos.kind_of? String
+        posnum = self.index(pos)
+        if posnum
+          b = self[0, posnum.to_i]
+          a = self[(posnum + pos.length)..-1]
+          self.replace([b, value, a].join(''))
+          return value
+        else
+          raise IndexError, "string not matched"
+        end
+      else
+        pos += self.length if pos < 0
+        if pos < 0 || pos > self.length
+          raise IndexError, "index #{args[0]} out of string"
```

Comment:
Yes, I figured :-)

On a somehow related note, I can't see where in the spec `String#[]` (15.2.10.5.6) is supposed to raise anything in those cases. :-/